common/concurrency/patterns.h

```diff
@@ -0,0 +1,43 @@
+#ifndef SORBET_CONCURRENCY_PATTERNS_H
+#define SORBET_CONCURRENCY_PATTERNS_H
+
+#include "common/concurrency/ConcurrentQueue.h"
+#include "common/concurrency/WorkerPool.h"
+
+namespace sorbet {
+
+class ConcurrencyPatterns {
```

Comment:
I think I'm fine with this change overall but I'm not sure I'm in love with the name of `Patterns` but in the grand scheme of things that probably doesn't matter.

I defer to you and @froydnj on this PR.

Reply:
I don't love this name either, and would love a better one :)

Reply:
Another part of my problem is that the Concurrency is the implementation detail. The API is parallel programming.

I would be willing to have something like class Parallel

Parallel::iterate

Reply:
I like that name much more!